src/imageio/format/jpeg.c

```diff
@@ -477,7 +534,24 @@ void gui_init(dt_imageio_module_format_t *self)
   dt_bauhaus_slider_set(g->quality, dt_conf_get_int("plugins/imageio/format/jpeg/quality"));
   gtk_box_pack_start(GTK_BOX(box), GTK_WIDGET(g->quality), TRUE, TRUE, 0);
   g_signal_connect(G_OBJECT(g->quality), "value-changed", G_CALLBACK(quality_changed), NULL);
-  // TODO: add more options: subsample dreggn
+
+  DT_BAUHAUS_COMBOBOX_NEW_FULL(g->subsample,
```

Comment:
If you wanted to, you could define the options in `darktableconfig.xml.in` and use `dt_gui_preferences_enum` to create the widget. The advantage would be no need for a "combobox_changed" function or to initialise with the existing value and the preference would be saved in darktablerc as a string rather than an int (so the order/list can be amended later) and there will be input validation if the user manually changed it.

Reply:
Thanks for the idea!

As for the ability to change the order of menu items, in general, this is a very nice feature. However, as for the specific list of subsampling options, I am sure that possible additions to the current list in the future will not cause problems. The point is that this list should obviously be in descending order of chroma resolution, and now it contains all the options with the highest resolution. Therefore, in practice, the only possible change to the list is the addition of even smaller resolutions at the end of the list, which will not harm the number of the menu item already saved in darktablerc.

It's also not very clear what we should store in the module's parameter structure instead of the option number in the subsampling menu.

So the bottom line, the benefits are very attractive, but I don't think it makes sense to do it right now.

Reply:
> I don't think it makes sense to do it right now.

Fully understood and expected and totally up to you.

Of course you do realise that making a change later, when everybody has ints in their darktablerc is much less trivial. Which is why I've not bothered to convert any of the current uses of `DT_BAUHAUS_COMBOBOX_NEW_FULL` in imageio.

Reply:
> Which is why I've not bothered to convert any of the current uses of `DT_BAUHAUS_COMBOBOX_NEW_FULL` in imageio.

That's exactly what I thought when I saw that no module in iop and imageio uses the new method.

> making a change later, when everybody has ints in their darktablerc is much less trivial.

It's obvious, but as I wrote above in a long paragraph I don't have a lot of motivation to spend time on switching to a new method right now (the code freeze is coming soon, and I'm still hoping to catch up with one more feature) as I'm sure that in this particular case there will be no need to _reorder_ the list of subsamples in the future.

But since you are much better at it (and I still need time to learn it), it would be great if you could make the changes you suggested after accepting this PR.